File: app/src/main/java/com/example/vaccineapp/model/InfoElement.java
package com.example.vaccineapp.model;

import com.example.vaccineapp.R;

public class InfoElement {

        private String information;
        private int alertLevel;

        public InfoElement(String information, int alertLevel){
            this.information = information;
            this.alertLevel = alertLevel;
        }

        public String getInformation() {
            return information;
        }

        public int getAlert() {
            return alertLevel;
        }

        public int getAlertResource(){
            if(this.alertLevel == 1){
                return R.drawable.check;
            }else if(this.alertLevel == -1){
                return R.drawable.warning_yellow;
            }else if(this.alertLevel == -2){
                return R.drawable.warning_red;
            }else{
                return R.drawable.warning_yellow;
            }
        }

        public void checkInfo(){
            this.alertLevel = 1;
        }
}

File: app/src/main/java/com/example/vaccineapp/model_kotlin/InfoElement.kt
package com.example.vaccineapp.model_kotlin

import com.example.vaccineapp.R

const val OK = 1
const val WARNING = -1
const val DANGER = -2


class InfoElement (val information: String, var alertLevel: Int){

    fun getAlertResource(): Int{
        return when(alertLevel){
            OK -> R.drawable.check
            WARNING -> R.drawable.warning_yellow
            DANGER -> R.drawable.warning_red
            else -> R.drawable.check
        }
    }

    fun checkInfo(){
        alertLevel = 1
    }





}